src/routes/(app)/order/[id]/+page.server.ts

```diff
@@ -97,5 +100,45 @@ export const actions = {
 		);
 
 		throw redirect(303, request.headers.get('referer') || '/');
+	},
+	addPayment: async ({ params, request, locals }) => {
+		const order = await collections.orders.findOne({
+			_id: params.id
+		});
+
+		if (!order) {
+			throw error(404, 'Order not found');
+		}
+
+		if (order.status !== 'pending') {
+			throw error(400, 'Order is not pending');
+		}
+
+		let methods = paymentMethods({ role: locals.user?.roleId });
+
+		for (const item of order.items) {
+			if (item.product.paymentMethods) {
+				methods = methods.filter((method) => item.product.paymentMethods?.includes(method));
+			}
+		}
+
+		if (!methods.length) {
+			throw error(400, 'No payment methods available');
+		}
+
+		const formData = await request.formData();
+		const parsed = z
+			.object({
+				method: z.enum(methods as [PaymentMethod, ...PaymentMethod[]])
+			})
+			.parse({
+				method: formData.get('method')
+			});
+
+		await addOrderPayment(order, parsed.method, order.currencySnapshot.main.totalPrice, {
+			expiresAt: addMinutes(new Date(), runtimeConfig.desiredPaymentTimeout)
+		});
```

Comment:
For `expiresAt`, to be consistent, wse should use `paymentMethodExpiration` as in https://github.com/be-BOP-io-SA/be-BOP/blob/fc5bceb9b3b01e0d4fb117c0f58ee764ab83be4b/src/lib/server/orders.ts#L995-L998

For the amount, it should be the same as the cancelled payment. 

In fact it should be a replace payment call, like in https://github.com/be-BOP-io-SA/be-BOP/pull/1717, so the url should be something like `order/[id]/payment/[paymentId]/+page.server.ts`

And the user needs to provide the id of the failed payment.

The reason we can't just put the full amount, is that for example there are orders with deposits.